File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
full_colors = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']

def label(colors):
    first_index = full_colors.index(colors[0])
    second_index = full_colors.index(colors[1])
    third_index = full_colors.index(colors[2])
    
    number = int(str(first_index) + str(second_index))
    power = 10 ** third_index
    value = number * power

    level = 0
    unit = ''
    while value % 1000 == 0 and value > 0:
        value = value // 1000
        level += 1

    if level == 1:
        unit = 'kiloohms'
    elif level == 2:
        unit = 'megaohms'
    elif level == 3:
        unit = 'gigaohms'
    else:
        unit = 'ohms'
    
    return ' '.join([str(value), unit])
    

File: solutions/typescript/resistor-color-trio/1/resistor-color-trio.ts
export function decodedResistorValue(colors: string[]): string {
  const first = COLORS.indexOf(colors[0]);
  const second = COLORS.indexOf(colors[1]);
  const multiple = 10 ** COLORS.indexOf(colors[2])
  const result = (first * 10 + second) * multiple;

  if(result >= 1_000_000_000) {
    return result / 1_000_000_000 + ' gigaohms';
  }
  if(result >= 1_000_000) {
    return result / 1_000_000 + ' megaohms';
  }
  if(result >= 1_000) {
    return result / 1_000 + ' kiloohms';
  }

  return result + ' ohms';
}

export const COLORS: string[] = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
]